specs/interop/predeploys.md

```diff
@@ -196,6 +198,28 @@ function executeMessage(Identifier calldata _id, address _target, bytes calldata
 
 Note that the `executeMessage` function is `payable` to enable relayers to earn in the gas paying asset.
 
+An example of a custom entrypoint utilizing `validateMessage` for cross-chain messages
+
+```solidity
+contract MyCrossChainApp {
+    event MyCrossChainEvent();
+
+    function relayMessage(Identifier calldata _id, bytes calldata _msg) public {
+        // Example app-level validation
+        //  - Expected event via the selector
+        //  - Assertion on the expected emitter of the event
+        require(MyCrossChainEvent.selector == _msg[:4]);
```

Comment:
Why observe only the first four bytes? An event selector is 32 bytes

Reply:
ahh yea my mistake